src/build/__init__.py

```diff
@@ -13,23 +13,29 @@
 import difflib
 import logging
 import os
-import re
 import subprocess
 import sys
-import textwrap
-import types
 import warnings
 import zipfile
 
-from collections import OrderedDict
-from collections.abc import Iterator, Set
-from typing import Any, Callable, Mapping, MutableMapping, Optional, Sequence, Tuple, Type, Union
+from collections.abc import Iterator
+from typing import Any, Callable, Mapping, Optional, Sequence, TypeVar, Union
 
 import pyproject_hooks
 
+from . import env
+from ._exceptions import (
```

Comment:
Does this need to be `_exceptions` instead of `exceptions`? Exceptions are not private, and putting them in a common place doesn't seem to be API we need to hide (the tracebacks will print this module name unless you set `__module__` manually). Reexporting them here is fine, but hiding the path to them doesn't seem necessary. But don't feel strongly.

Reply:
I didn't wanna add another place where they can be publicly imported from in this PR, but I'm supportive of making `_exceptions` public.

Reply:
Okay, fine for a followup then, renaming this & can be discussed there.